### PR TITLE
Combined TextField(Input) and TextArea implementation for shared focus()

### PR DESCRIPTION
**Pull request by swese44**
- Implements public `.focus()` on TextArea
- Combines debouncing and other functionality in TextField and TextArea to `components/internal/TextField`

Note that i plan to rename TextField to TextInput in another PR

## Pull request checklist

* [x] Component `README.md` file is up-to-date.
* [x] Component is unit tested.

### CODE REVIEW

File: src/components/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -1,45 +1,18 @@
...
-   * Height of the field, in number of rows.
-   */
-  rows?: number;
-
-  /**
-   * Whether or not to auto adjust textField height.
-   */
-  autoAdjustHeight?: boolean;
-}
+export type TextAreaProps = Omit<InternalTextAreaProps, 'multiline'>;
```
Comment (swese44) [inline src/components/TextArea/TextArea.tsx]: i've used `Omit` to remove the `multiline` prop from the public TextFieldProps and TextAreaProps interfaces while the TextField and TextArea component manually set it to true/false. this way the underlying `internal/TextField` component has type safety around multiline (TextArea) vs. single line (TextInput), and the consumer app/engineer can just use the public TextArea or TextInput component without knowing the implementation

File: src/components/TextArea/TextArea.test.tsx [hunk 2]
```diff
@@ -51,6 +53,7 @@ describe('<TextArea />', () => {
   describe('with description and no error', () => {
     beforeEach(() => {
       component = shallow(<TextArea description="DESCRIPTION" />)
+        .dive()
         .dive()
         .dive();
```
Comment (unindented) [inline src/components/TextArea/TextArea.test.tsx]: commander_keen_4_-_dos_-_dope_fish

Comment (swese44) [reply src/components/TextArea/TextArea.test.tsx]: lol

File: src/components/TextArea/TextArea.tsx [hunk 3]
```diff
@@ -1,45 +1,18 @@
...
 
 /**
- * The TextArea component enables a user to type text into an app. The text
- * displays on the screen in a simple, uniform format.
+ * TextArea is rendered as an html textarea tag, used to allow more than one line of text to be entered.
  */
-class TextArea extends React.Component<TextAreaProps & DebouncedOnChangePrivateProps> {
+export default class TextField extends React.Component<TextAreaProps> {
```
Comment (unindented) [inline src/components/TextArea/TextArea.tsx]: Should this say `class TextArea`?

Comment (swese44) [reply src/components/TextArea/TextArea.tsx]: yes it should

File: src/components/TextArea/__snapshots__/TextArea.test.tsx.snap [hunk 4]
```diff
@@ -6,7 +6,8 @@ exports[`<TextArea /> with all options matches its snapshot 1`] = `
   ariaLabel="ARIA_LABEL"
   autoAdjustHeight={true}
   borderless={false}
-  className="y-base-text-field y-text-area CLASS"
+  className="y-textField y-textArea CLASS"
```
Comment (unindented) [inline src/components/TextArea/__snapshots__/TextArea.test.tsx.snap]: Does `y-textField` have the base text field styles? Should it have `base` in the name?

File: src/components/TextArea/TextArea.tsx (hunk 3 above)
Comment (swese44) [reply src/components/TextArea/TextArea.tsx]: updated

File: src/components/TextArea/__snapshots__/TextArea.test.tsx.snap (hunk 4 above)
Comment (swese44) [reply src/components/TextArea/__snapshots__/TextArea.test.tsx.snap]: the "TextField" class is the internal component. after this i'll follow up with a PR to rename the public TextField component to TextInput for a bit more clarity
https://www.yammer.com/microsoft.com/#/Threads/show?threadId=1085910906

**Pull request merged**